packages/rich-text/src/component/index.js

```diff
@@ -137,10 +137,12 @@ export function useRichText( {
 
 		// Selection must be updated first, so it is recorded in history when
 		// the content change happens.
-		onSelectionChange( start, end );
-		onChange( _value.current, {
-			__unstableFormats: formats,
-			__unstableText: text,
+		wp.data.batch( () => {
+			onSelectionChange( start, end );
+			onChange( _value.current, {
+				__unstableFormats: formats,
+				__unstableText: text,
+			} );
```

Comment:
Would `unstable_batchedUpdates` work in this context? And I believe starting from React 18, it will also become the default for all state updates. Might worth to just use the official API React provides. There's also a `react-redux` version: [`batch`](https://react-redux.js.org/api/batch), which is basically the same as `unstable_batchedUpdates`.

Reply:
It's not just about React rendering though, calling a dispatch (data store) triggers all the `useSelect` listeners (calling selectors) even before the `setState` happens. In most cases `setState` don't happen at all because the selectors return the same values (shallow comparison).

Reply:
I made some tests here locally. 

 - with `unstable_batchedUpdates`, I see a `5ms` improvement
 - with `wp.data.batch` I see a `10ms` improvement
 - with both it's still `10ms` improvement.